docs/plugin-developer-guide/documentation/README.md

```diff
@@ -0,0 +1,161 @@
+                "tasks:",
+                "- id: update",
+                "  type: io.kestra.plugin.jdbc.postgresql.Query",
+                "  url: jdbc:postgresql://127.0.0.1:56982/",
+                "  username: postgres",
+                "  password: pg_passwd",
+                "  sql: select concert_id, available, a, b, c, d, play_time, library_record, floatn_test, double_test, real_test, numeric_test, date_type, time_type, timez_type, timestamp_type, timestampz_type, interval_type, pay_by_quarter, schedule, json_type, blob_type from pgsql_types",
+                "  fetch: true"}
+        )
+    }
+)
+```
+
+For convenience, the `code` attribute of the `@Example` annotation is a list of strings, each string will be a line of example. This avoid concatenating multi-line strings in a single attribute.
+
+You can add multiple examples if needed.
+
+### Document the plugin properties
+
+In a plugin, all properties must be annotated by `io.kestra.core.models.annotations.PluginProperty` and should provide documentation via the `io.swagger.v3.oas.annotations.media.Schema` annotation and validation rules via `javax.validation.constraints.*`.
+
+For validation rules, please check the [Properties validation](../runnable-task/#properties-validation) documentation page.
+
+The `@PluginProperty` annotation contains two attributes:
+
+- `dynamic`: set it to true if the property will be rendered dynamically, see [Render Variable](..//runnable-task/#render-variables).
+- `additionalProperties`: you can use it to denote the sub-type of the property, for example when using a `Map<String, Message>` you can set it to `Message.class`.
+
+The Swagger `@Schema` annotation contains a lot of attributes that can be used to document the plugin properties, the most useful are:
+
+- `title`: short description of a property.
+- `description`: long description of a property.
+- `anyOf`: allows to specify which types are allowed when the type of the property can be extended (interface, abstract classes, ...). This should be set when the type of the property is `Object`.
+
+The `@Schema` and `@PluginProperty` annotations can be used on fields, methods or classes.
+
+A lot of tasks can take input from multiple sources on the same property, they usually have a single `from` property that can be a string that represent a file in the Kestra Storage, a single object or a list of objects. To document such property you can use `anyOf` this way:
+
+```java
+@PluginProperty(dynamic = true)
+@NotNull
+@Schema(
+    title = "The source of the published data.",
+    description = "Can be an internal storage URI, a list of Pub/Sub messages, or a single Pub/Sub message.",
+    anyOf = {String.class, Message[].class, Message.class}
+)
+private Object from;
+```
+
+Note that, due to limitation on the way JSON schema works, you cannot add `@Schema` both on a Java enum type and the plugin property that uses this type. We advise to avoid using `@Schema` on enumerations.
```

Comment:
```suggestion
**Note:** Due to limitations on how JSON schema works, you cannot add `@Schema` on a Java enum type and the plugin property that uses this type. We advise avoiding using `@Schema` on enumerations.
```

Reply:
I'm pretty sure we have a format to display notes as a special visual block, I'll check and update the formating.